src/libcore/iter.rs

```diff
@@ -2224,13 +2222,11 @@ impl<I> DoubleEndedIterator for Fuse<I> where I: DoubleEndedIterator {
         if self.done {
             None
         } else {
-            match self.iter.next_back() {
-                None => {
-                    self.done = true;
-                    None
-                }
-                x => x
+            let next = self.iter.next_back();
+            if next.is_none() {
+                self.done = true;
```

Comment:
Anytime I see a `done` variable I instantly go to "this is a one-transition boolean", personally.